test/test_cuda.py

```diff
@@ -2075,6 +2075,31 @@ def _worker(t):
             for t in range(num_threads):
                 self.assertEqual(results[t].sum().item(), size * size)
 
+    def testCudaRecordStreamWithWrongDevice(self):
+        t = torch.randn(5, 5)
+        with self.assertRaisesRegex(RuntimeError, "Tensor is not allocated in CUDA"):
+            t.record_stream(torch.cuda.current_stream())
```

Comment:
Let's also test and make sure that the error throws on a new CUDA stream.

Reply:
Discussed with Shen, it is already covered by other test cases in this file